### PR TITLE
Add button to remove mod metadata

### LINKED ISSUE #1528: Add the ability to delete metadata per-mod or for all mods in an instance
### Role

I host a modded Minecraft server with a sizeable modpack, the launcher is used for easing the downloading and updating of mods and keeping backup instances for each major version of the pack.

### Suggestion

Add the ability to delete metadata for mods in each instance. This should be either per-mod or for all mods in the instance at once. This could be included as a button on the right side of the interface and/or as a right click option on each mod in the list.

### Benefit

If a mod has moved development to strictly modrinth or curseforge it would be beneficial to be able to easily reset the metadata for a mod so that you can select it's appropriate download location again. It could also be useful if the incorrect download source was selected by accident and it needs to be switched.

### This suggestion is unique

- [X] I have searched the issue tracker and did not find an issue describing my suggestion, especially not one that has been rejected.

### You may use the editor below to elaborate further.

This can currently be accomplished by manually going in to the mods folder and deleting files in the .index folder for each mod but this invites a higher chance of user error and is not intuitive to many users who are unfamiliar with the location or purpose of these files.

### PR DESCRIPTION
**Pull request by Trial97**
fixes #1528

### CODE REVIEW

File: launcher/ui/pages/instance/ModFolderPage.cpp [hunk 1]
```diff
@@ -297,3 +306,27 @@ void ModFolderPage::visitModPages()
...
+    if (selectionCount == 0)
+        return;
+    else if (selectionCount > 1)
+        text = tr("You are about to remove the metadata for %1 mods.\n"
+                  "Are you sure?")
+                   .arg(selectionCount);
+    else
+        text = tr("You are about to remove the metadata for %1.\n"
+                  "Are you sure?")
+                   .arg(m_model->at(selection.at(0).row())->name());
```
Comment (TheKodeToad) [inline launcher/ui/pages/instance/ModFolderPage.cpp]: I don't think a confirmation is *necessarily* needed with one mod.

File: launcher/ui/pages/instance/ExternalResourcesPage.ui [hunk 2]
```diff
@@ -168,6 +168,17 @@
...
   </action>
+  <action name="actionRemoveItemMetadata">
+   <property name="enabled">
+    <bool>false</bool>
+   </property>
+   <property name="text">
+    <string>Remove metadata</string>
+   </property>
+   <property name="toolTip">
+    <string>Remove selected mod metadata</string>
```
Comment (TheKodeToad) [inline launcher/ui/pages/instance/ExternalResourcesPage.ui]:
```suggestion
    <string>Remove mod's metadata</string>
```

**Review by TheKodeToad — CHANGES_REQUESTED** [review]

File: launcher/ui/pages/instance/ModFolderPage.cpp [hunk 3]
```diff
@@ -92,6 +92,10 @@ ModFolderPage::ModFolderPage(BaseInstance* inst, std::shared_ptr<ModFolderModel>
         ui->actionsToolbar->addAction(ui->actionVisitItemPage);
         connect(ui->actionVisitItemPage, &QAction::triggered, this, &ModFolderPage::visitModPages);
 
+        ui->actionRemoveItemMetadata->setToolTip(tr("Remove mod's metadata"));
+        ui->actionsToolbar->insertActionAfter(ui->actionRemoveItem, ui->actionRemoveItemMetadata);
```
Comment (TheKodeToad) [inline launcher/ui/pages/instance/ModFolderPage.cpp:96]:
```suggestion
        ui->actionsToolbar->addAction(ui->actionRemoveItemMetadata);
```
oops, I missed a bit lol

Comment (Trial97) [reply launcher/ui/pages/instance/ModFolderPage.cpp:96]: I would want it to be near Remove or at least before view folder actions

Comment (TheKodeToad) [reply launcher/ui/pages/instance/ModFolderPage.cpp:96]: Sorry - oops

**Review by TheKodeToad — APPROVED** [review]

**Review by DioEgizio — APPROVED** [review]

**Review by Scrumplex — APPROVED** [review]

**Pull request merged**